librz/core/analysis_tp.c

```diff
@@ -466,9 +490,19 @@ void free_op_cache_kv(HtUPKv *kv) {
 	rz_analysis_op_free(kv->value);
 }
 
-void handle_stack_canary(RzCore *core, Sdb *trace, RzAnalysisOp *aop, int cur_idx) {
-	const char *query = sdb_fmt("%d.addr", cur_idx - 1);
-	ut64 mov_addr = sdb_num_get(trace, query, 0);
+void handle_stack_canary(RzCore *core, RzAnalysisOp *aop, int cur_idx) {
+	RzILTraceInstruction *prev_trace =
+		rz_analysis_esil_get_instruction_trace(
+			core->analysis->esil->trace,
+			cur_idx - 1);
+	ut64 mov_addr;
+
+	if (prev_trace) {
+		mov_addr = prev_trace->addr;
+	} else {
+		mov_addr = UT64_MAX;
+	}
```

Comment:
shall we exit if we don't have a `prev_trace`? I'm not sure what we expect to find at address UT64_MAX.

Reply:
It's an error mark for `rz_core_analysis_op`, when hit UT64_MAX it would return a NULL and skip the following analysis. The behavior is the same as we exit from this function.